### PR TITLE
PP-11563 Separate WalletPaymentInfo for Apple and Google

### PR DESCRIPTION
**Pull request by stephencdaly**
Requests to the authorise endpoints for apple and google pay contain some common data, but some different data in the `payment_info` field.

Make this clear by adding ApplePayPaymentInfo, which exists on the ApplePayAuthRequest and GooglePayPaymentInfo which exists on WorldpayGooglePayAuthRequest and StripeGooglePayAuthRequest.

Make both these classes extend the abstract class WalletPaymentInfo which contains shared fields and allows us to share some code for using this base class.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayAuthRequest.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import io.sentry.protocol.App;
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayAuthRequest.java]: is this being used?

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayAuthRequest.java]: dunno how that got in there!

File: src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayPaymentInfo.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package uk.gov.pay.connector.wallets.applepay.api;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import uk.gov.pay.connector.gateway.model.PayersCardType;
+import uk.gov.pay.connector.wallets.model.WalletPaymentInfo;
+
+public class ApplePayPaymentInfo extends WalletPaymentInfo {
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayPaymentInfo.java:7]: probably being pedantic here but could this be `ApplePayWalletPaymentInfo` instead?

File: src/main/java/uk/gov/pay/connector/wallets/googlepay/api/GooglePayPaymentInfo.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package uk.gov.pay.connector.wallets.googlepay.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+import uk.gov.pay.connector.gateway.model.PayersCardType;
+import uk.gov.pay.connector.wallets.model.WalletPaymentInfo;
+
+import java.util.Optional;
+
+public class GooglePayPaymentInfo extends WalletPaymentInfo {
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/wallets/googlepay/api/GooglePayPaymentInfo.java:10]: `GooglePayWalletPaymentInfo`

File: src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayPaymentInfo.java (hunk 2 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayPaymentInfo.java:7]: Hmm not sure I see the need - Apple Pay is a type of wallet. I don't think it's a thing that subclasses need to include the name of the base class in its entirety

Comment (nlsteers) [reply src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayPaymentInfo.java:7]: subclass naming is subjective I guess 😅

File: src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayAuthRequest.java (hunk 1 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/wallets/applepay/api/ApplePayAuthRequest.java]: removed

**Review by nlsteers — APPROVED** [review]
Looks good, nice work 💯

**Pull request merged**